xcodebuild: WARNING: Using the first of multiple matching destinationstore

### I have searched and made sure there are no existing issues for the issue I am filing

- [X] I have searched the existing issues

### Description

I try to build a release for the appstore with XCode 14.2 and Ti SDK 12.0.0.GA. The build succeeds but once the xcodebuild is fired I got the following warning:

_[ERROR] --- xcodebuild: WARNING: Using the first of multiple matching destinations:
[ERROR] { platform:iOS, id:dvtdevice-DVTiPhonePlaceholder-iphoneos:placeholder, name:Any iOS Device }
[ERROR] { platform:iOS Simulator, id:dvtdevice-DVTiOSDeviceSimulatorPlaceholder-iphonesimulator:placeholder, name:Any iOS Simulator Device }
...
[ERROR] { platform:iOS Simulator, id:A79DB9D7-CF8D-4877-8E0F-0DC4837641E3, OS:16.2, name:iPhone SE (3rd generation) }
[ERROR] ** ARCHIVE FAILED **_

Searching within the issues I found [13175](https://github.com/tidev/titanium_mobile/issues/13175) which looks like the same, but doesn't fix the dist-appstore target. I fixed the issue by adding 'dist-appstore' to [iphone/cli/commands/_build.js](https://github.com/tidev/titanium_mobile/pull/13172/files#diff-102dc6db94a39ec27245ad857eb14114c62d7dc3508f5207bd28e05bd335f398R7105) as mentioned for device and dist-adhoc in  #13175.

Here is my env spec:

**Node.js**
Node.js Version             = 16.19.1
npm Version                 = 8.19.3

**Titanium CLI**
CLI Version                 = 6.1.1
node-appc Version           = 1.1.6

**Xcode**
14.2 (build 14C18) - Xcode default
iOS SDKs                  = 16.2
iOS Simulators            = 16.2
Watch SDKs                = 9.1
Watch Simulators          = none
Supported by TiSDK 12.0.0.GA = yes
EULA Accepted             = yes
Teams                     = none


### Expected Behavior

dist-appstore build should succeed.

### Actual behavior

dist-appstore build fails

### Reproducible sample

Create a sample app and use my environment listed within the description. 

### Steps to reproduce

Run 

`titanium build -p ios -T dist-appstore -O dist -b`

### Platform

iOS

### SDK version you are using

12.0.0.GA

### Alloy version you are using

2.0.2

--- a/iphone/cli/commands/_build.js
+++ b/iphone/cli/commands/_build.js
@@ -7109,8 +7109,10 @@ iOSBuilder.prototype.invokeXcodeBuild = async function invokeXcodeBuild(next) {
 			this.logger.warn(`The app is using native modules (${Array.from(this.legacyModules)}) that do not support arm64 simulators, we will exclude arm64. This may fail if you're on an arm64 Apple Silicon device.`);
 			args.push('EXCLUDED_ARCHS=arm64');
 		}
-	} else if (this.target === 'device' || this.target === 'dist-adhoc') {
+	} else if (this.target === 'device' || this.target === 'dist-adhoc' || this.target === 'dist-appstore') {
 		args.push('-destination', 'generic/platform=iOS');
+	} else if (this.target === 'macos' || this.target === 'dist-macappstore') {
+		args.push('-destination', 'generic/platform=macOS');
 	}
 
 	xcodebuildHook(

